src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts

```diff
@@ -419,6 +432,47 @@ export class CourseProgressComponent implements OnInit, OnDestroy, AfterViewInit
         this.populateCourseDashboardData();
       });
   }
+  getReportUpdatedOnDate() {
+    const batchId = _.get(this.queryParams, 'batchIdentifier');
+    const reportParams = {
+      'course-progress-reports': `course-progress-reports/report-${batchId}.csv`,
+      'assessment-reports': `assessment-reports/report-${batchId}.csv`
+    };
+    const url = `${this.config.urlConFig['URLS'].COURSE.GET_REPORTS_METADATA}`;
+    const requestParams = {
+      params: {
+        fileNames: JSON.stringify(reportParams)
+      }
+    };
+    this.usageService.getData(url, requestParams).subscribe((response: any) => {
+      if (_.get(response, 'responseCode') === 'OK') {
+        const courseProgressReportData = _.get(response, `result.course-progress-reports.lastModified`);
+        const assessmentReportData = _.get(response, `result.assessment-reports.lastModified`);
+        this.progressReportUpdatedOn =  courseProgressReportData ? moment(courseProgressReportData).format('DD/MM/YYYY') : '';
+        this.scoreReportUpdatedOn = assessmentReportData ? moment(assessmentReportData).format('DD/MM/YYYY') : '';
+        const telemetryEventObject = [];
+        for (const reportName of Object.keys(reportParams)) {
+          const event = {
+            context: {
+              env: this.activatedRoute.snapshot.data.telemetry.env
+            },
+            edata: {
+              type: this.activatedRoute.snapshot.data.telemetry.type,
+              level: 'SUCCESS',
+              // tslint:disable-next-line: max-line-length
+              message: _.get(response, `result.${reportName}.lastModified`) !== '' ? `${reportName} is available` : `${reportName} is not available`,
+              pageid: this.activatedRoute.snapshot.data.telemetry.pageid
+            }
+          };
+          telemetryEventObject.push(event);
+        }
+        if (telemetryEventObject.length === 2) {
```

Comment:
This logic should go into service. When the API is Succes/Fail then it should log telemetry event.